From buytenh@wantstofly.org  Tue May 20 10:41:21 2008
From: Lennert Buytenhek <buytenh@wantstofly.org>
Date: Tue, 20 May 2008 19:08:53 +0200
Subject: USB: ehci-orion: the Orion EHCI root hub does have a Transaction Translator
To: Greg Kroah-Hartman <gregkh@suse.de>, David Brownell <david-b@pacbell.net>
Cc: "Rafael J. Wysocki" <rjw@sisk.pl>, Alan Stern <stern@rowland.harvard.edu>, Andrew Morton <akpm@linux-foundation.org>, Nicolas Pitre <nico@cam.org>
Message-ID: <20080520170853.GA26577@xi.wantstofly.org>
Content-Disposition: inline


Commit 7329e211b987a493cbcfca0e98c60eb108ab42df ("USB: root hubs don't
lie about their number of TTs") requires the various platform EHCI
glue modules to set ->has_tt if the root hub has a Transaction
Translator.

The Orion EHCI root hub does have a Transaction Translator, so set
->has_tt in ehci_orion_setup().  This fixes oopsing on plugging in a
low speed device.

Signed-off-by: Lennert Buytenhek <buytenh@marvell.com>
Acked-by: Nicolas Pitre <nico@marvell.com>
Acked-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ehci-orion.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/host/ehci-orion.c
+++ b/drivers/usb/host/ehci-orion.c
@@ -115,6 +115,8 @@ static int ehci_orion_setup(struct usb_h
 	if (retval)
 		return retval;
 
+	hcd->has_tt = 1;
+
 	ehci_reset(ehci);
 	ehci_port_power(ehci, 0);
 
